buildpacks/ruby/tests/integration_test.rs

```diff
@@ -11,22 +11,67 @@ use std::thread;
 use std::time::{Duration, Instant};
 use ureq::Response;
 
+// Test that:
+// - Cached data "stack" is preserved and will be successfully migrated to "targets"
 #[test]
 #[ignore = "integration test"]
-fn test_default_app() {
+fn test_migrating_metadata() {
+    let builder = "heroku/builder:22";
+    let app_dir = "tests/fixtures/default_ruby";
+
+    TestRunner::default().build(
+        BuildConfig::new(builder, app_dir).buildpacks([BuildpackReference::Other(
+            "docker://docker.io/heroku/buildpack-ruby:2.1.2".to_string(),
+        )]),
+        |context| {
+            println!("{}", context.pack_stdout);
+            context.rebuild(
+                BuildConfig::new(builder, app_dir).buildpacks([BuildpackReference::CurrentCrate]),
+                |rebuild_context| {
+                    println!("{}", rebuild_context.pack_stdout);
+
+                    assert_contains!(rebuild_context.pack_stdout, "Using cached Ruby version");
+                    assert_contains!(rebuild_context.pack_stdout, "Loading cached gems");
+                },
+            );
+        },
+    );
+}
+
+#[test]
+#[ignore = "integration test"]
+fn test_default_app_ubuntu20() {
+    TestRunner::default().build(
+        BuildConfig::new("heroku/builder:20", "tests/fixtures/default_ruby"),
+        |context| {
+            println!("{}", context.pack_stdout);
+            assert_contains!(context.pack_stdout, "# Heroku Ruby Buildpack");
+            assert_contains!(
+                context.pack_stdout,
+                r#"`BUNDLE_BIN="/layers/heroku_ruby/gems/bin" BUNDLE_CLEAN="1" BUNDLE_DEPLOYMENT="1" BUNDLE_GEMFILE="/workspace/Gemfile" BUNDLE_PATH="/layers/heroku_ruby/gems" BUNDLE_WITHOUT="development:test" bundle install`"#);
+
+            assert_contains!(context.pack_stdout, "Installing webrick");
+        },
+    );
+}
+
+#[test]
+#[ignore = "integration test"]
+fn test_default_app_latest_distro() {
     TestRunner::default().build(
         BuildConfig::new("heroku/builder:22", "tests/fixtures/default_ruby"),
         |context| {
+            println!("{}", context.pack_stdout);
```

Comment:
(optional) Fwiw the `assert_contains!` macro prints the full original string, so this line is redundant, given the assertions below it will print the full string anyway.

Reply:
I added it because of the later `rebuild` calls and the following assert_contains which would only show you the rebuilt stdout instead of all of it. 

So if the first build had something off or unexpected but worked and then the second build is where it actually failed, it will give you all of the context. I would like to leave it in.